tests/dataimportcron_test.go

```diff
@@ -0,0 +1,210 @@
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+	cdiclientset "kubevirt.io/containerized-data-importer/pkg/client/clientset/versioned"
+	"kubevirt.io/containerized-data-importer/pkg/controller"
+	"kubevirt.io/containerized-data-importer/tests/framework"
+	"kubevirt.io/containerized-data-importer/tests/utils"
+)
+
+const (
+	dataImportCronTimeout = 90 * time.Second
+)
+
+var _ = Describe("DataImportCron", func() {
+	const (
+		scheduleEveryMinute = "* * * * *"
+		scheduleOnceAYear   = "0 0 1 1 *"
+	)
+	var (
+		f                = framework.NewFramework(namespacePrefix)
+		registryPullNode = cdiv1.RegistryPullNode
+		//trustedRegistryURL = func() string { return fmt.Sprintf(utils.TrustedRegistryURL, f.DockerPrefix, f.DockerTag) }
+		cron *cdiv1.DataImportCron
+		err  error
+	)
+
+	AfterEach(func() {
+		if cron != nil {
+			By("Delete cron")
+			err = DeleteDataImportCron(f.CdiClient, cron.Namespace, cron.Name)
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	table.DescribeTable("should", func(schedule string, repeat int) {
+		//FIXME: skopeo inspect docker://registry:5000/cdi-func-test-tinycore:latest": exit status 1:
+		//       time="2021-11-07T11:51:22Z" level=fatal msg="Error parsing image name \"docker://registry:5000/cdi-func-test-tinycore:latest\":
+		//       error pinging docker registry registry:5000: Get \"https://registry:5000/v2/\": http: server gave HTTP response to HTTPS client
+		//url := trustedRegistryURL()
+		url := "docker://quay.io/kubevirt/fedora-cloud-registry-disk-demo:latest"
+
+		cron = NewDataImportCron("cron-test", "5Gi", schedule, "datasource-test", cdiv1.DataVolumeSourceRegistry{URL: &url, PullMethod: &registryPullNode})
+		By(fmt.Sprintf("Create new DataImportCron %s", url))
+		cron, err = CreateDataImportCronFromDefinition(f.CdiClient, f.Namespace.Name, cron)
+		Expect(err).ToNot(HaveOccurred())
+
+		if schedule == scheduleEveryMinute {
+			var cronjob *v1beta1.CronJob
+			By("Verify cronjob was created")
+			Eventually(func() bool {
+				cronjob, err = f.K8sClient.BatchV1beta1().CronJobs(f.CdiInstallNs).Get(context.TODO(), controller.GetCronJobName(cron), metav1.GetOptions{})
+				return err == nil && len(cronjob.Status.Active) > 0
+			}, dataImportCronTimeout, pollingInterval).Should(BeTrue())
+			Expect(err).ToNot(HaveOccurred())
+
+			By(fmt.Sprintf("Find pod of job %s", cronjob.Status.Active[0].Name))
+			Eventually(func() bool {
+				_, err = utils.FindPodByPrefix(f.K8sClient, f.CdiInstallNs, cronjob.Status.Active[0].Name, "")
+				return err == nil
+			}, dataImportCronTimeout, pollingInterval).Should(BeTrue())
+		}
+
+		var lastImportDv, currentImportDv string
+		for i := 0; i < repeat; i++ {
+			if repeat > 1 {
+				// Emulate source update
+				digest := []string{"sha256:0001112223334444", "sha256:0123456789abcdef", "sha256:aaabbbcccdddeeee"}[i%3]
```

Comment:
Don't really need i%3 if i is never > 3

Reply:
I tried import loops so it was helpful to be on the safe side. Removed.